### PR TITLE
Feature/mc 12537 edit role

### PR DESCRIPTION
**Pull request by katrinapoulin**
#### Changes made
- Added Edit operation for Roles in Kubernetes and Kubernetes extension

#### Related PRs
- [PR#213](https://github.com/cloudops/cloudmc-kubernetes-sdk/pull/213) in cloudmc-kubernetes-sdk

### CODE REVIEW

File: source/includes/kubernetes/_k8_roles.md [hunk 1]
```diff
@@ -169,6 +169,49 @@ Return value:
...
+```shell
+curl -X PUT \
+  -H "MC-Api-Key: your_api_key" \
+   "https://cloudmc_endpoint/v1/services/a_service/an_environment/roles/role-name/namespace-name?cluster_id=:cluster_id"
+  Content-Type: application/json
+  {
+    "metadata": {
+        "name": "role-name",
+        "namespace": "namespace-name"
+    }
```
Comment (danielvanacker) [inline source/includes/kubernetes/_k8_roles.md:185]: Might be good to add the Rules array object here as well

File: source/includes/kubernetes_extension/_k8_roles.md [hunk 2]
```diff
@@ -169,6 +169,50 @@ Return value:
...
+```shell
+curl -X PUT \
+  -H "MC-Api-Key: your_api_key" \
+   "https://cloudmc_endpoint/v1/services/a_service/an_environment/roles/role-name/namespace-name?cluster_id=:cluster_id"
+  Content-Type: application/json
+  {
+    "metadata": {
+        "name": "role-name",
+        "namespace": "namespace-name"
+    }
```
Comment (danielvanacker) [inline source/includes/kubernetes_extension/_k8_roles.md:185]: Same here

File: source/includes/kubernetes/_k8_roles.md [hunk 3]
```diff
@@ -169,6 +169,64 @@ Return value:
...
+<code>PUT /services/<a href="#administration-service-connections">:service_code</a>/<a href="#administration-environments">:environment_name</a>/roles/:id?cluster_id=:cluster_id</code>
+
+Replace a role in a given [environment](#administration-environments).
+
+| Required Attributes                 | &nbsp;                                                      |
+| ----------------------------------- | ----------------------------------------------------------- |
+| `metadata` <br/>_object_            | The metadata of the role.                                     |
+| `metadata.name` <br/>_string_       | The name of the role.                                         |
+| `metadata.namespace` <br/>_string_       | The namespace of the role.                                         |
+
```
Comment (danielvanacker) [inline source/includes/kubernetes/_k8_roles.md:227]: Add rules to optional attributes

File: source/includes/kubernetes_extension/_k8_roles.md [hunk 4]
```diff
@@ -169,6 +169,65 @@ Return value:
...
+<code>PUT /services/<a href="#administration-service-connections">:service_code</a>/<a href="#administration-environments">:environment_name</a>/roles/:id?cluster_id=:cluster_id</code>
+
+Replace a role in a given [environment](#administration-environments).
+
+| Required Attributes                 | &nbsp;                                                      |
+| ----------------------------------- | ----------------------------------------------------------- |
+| `metadata` <br/>_object_            | The metadata of the role.                                     |
+| `metadata.name` <br/>_string_       | The name of the role.                                         |
+| `metadata.namespace` <br/>_string_       | The namespace of the role.                                         |
+
```
Comment (danielvanacker) [inline source/includes/kubernetes_extension/_k8_roles.md:227]: same here

**Review by m-orfanos — APPROVED** [review]

File: source/includes/kubernetes/_k8_roles.md [hunk 5]
```diff
@@ -169,6 +169,69 @@ Return value:
...
+            "resource"
+         ],
+         "verbs": [
+            "get",
+            "watch",
+            "..."
+         ]
+      }
+   ]
+  }
```
Comment (Shabirmean) [inline source/includes/kubernetes/_k8_roles.md:201]: I think there is a missing "```" here

**Review by Shabirmean — APPROVED** [review]
LGTM!

**Pull request merged**